integration_tests/src/main/python/hybrid_parquet_test.py

```diff
@@ -290,6 +305,478 @@ def test_hybrid_parquet_filter_pushdown_timestamp(spark_tmp_path):
+           ('str3', StringGen(pattern='[a-z]{1,3}')),
+           ('digitstr', StringGen(pattern='[0-9]{1,5}')),
+           ('long', LongGen()),
+           ('comma_separated_str', StringGen(pattern='([0-9],){1,5}')),
+           ('json_str', StringGen(pattern=r'\{"a": "[a-z]{1,3}"\}')),
+]
+
+condition_list = [
+    # Boolean:
+    # Not
+    "not bool1",
+    # Boolean, Boolean:
+    # And, Or
+    "(bool1 and bool2)",
+    "(bool1 or bool2)",
+    # Double:
+    # Acos, Acosh, Asin, Asinh, Atan, Atan2, Atanh, Ceil, Cos, Cosh, IsNaN, Log, Log10, Log2, Rint, Sin, Sqrt, Tan, Tanh,Cbrt,Exp,Expm1,Floor,ToDegrees,ToRadians
+    "(acos(double1) > 0.5)", 
+    pytest.param("(acosh(double1) < 1.5)", marks=pytest.mark.xfail(reason='result not correct')),
+    "(asin(double1) > 0.2)", 
+    pytest.param("(asinh(double1) < 0.8)", marks=pytest.mark.xfail(reason='result not correct')), 
+    "(atan(double1) > 0.3)", 
+    "(atan2(double1, double2) > 0.4)", 
+    "(atanh(double1) < 0.6)",
+    "(ceil(double1) == 3)", 
+    "(cos(double1) < 0.7)", 
+    "(cosh(double1) > 0.9)",
+    "(isnan(double1)) ",
+    "(log(double1) > 1.0)", 
+    "(log10(double1) < 1.2)", 
+    "(log2(double1) > 1.4)", 
+    "(rint(double1) == 2)",
+    pytest.param("(sin(double1) > 0.1)", marks=pytest.mark.xfail(reason='not supported by gluten')),
+    pytest.param("(sqrt(double1) < 1.6)", marks=pytest.mark.xfail(reason='not supported by gluten')),
+    pytest.param("(tan(double1) > 0.8)", marks=pytest.mark.xfail(reason='not supported by gluten')),
+    pytest.param("(tanh(double1) < 0.4)", marks=pytest.mark.xfail(reason='not supported by gluten')),
+    pytest.param("(cbrt(double1) == 3.0)", marks=pytest.mark.xfail(reason='not supported by gluten')),
+    "(exp(double1) == 1.0)",
+    "(expm1(double1) == 1.0)",
+    "(floor(double1) == 1)",
+    pytest.param("(degrees(double1) == 0)", marks=pytest.mark.xfail(reason='not supported by gluten')),
+    pytest.param("(radians(double1) == 0)", marks=pytest.mark.xfail(reason='not supported by gluten')),
+    # Double, Int:
+    # Round,ArrayRepeat
+    "(round(double1) == 3)",
+    # Double, Double:
+    # NaNvl,Pow
+    "(nanvl(double1, 1) == 1)",
+    "(pow(double1, double2) == 2)",
+    # Double, Long:
+    # Round
+    "(round(double1, 1) == 3.1)",
+    # Timestamp:
+    # Hour, Minute
+    pytest.param("(hour(timestamp) == 1)", marks=pytest.mark.xfail(reason='timestamp filter not supported by gluten')),
+    pytest.param("(minute(timestamp) == 1)", marks=pytest.mark.xfail(reason='timestamp filter not supported by gluten')),
+    # Date:
+    # LastDay, WeekOfYear,DayOfMonth, DayOfWeek, DayOfYear, Month, Quarter, Second, UnixMicros, UnixMillis, UnixSeconds, WeekDay, Year
+    "(last_day(date1) == date1)",
+    "(weekofyear(date1) == 1)",
+    "(dayofmonth(date1) == 1)",
+    "(dayofweek(date1) == 1)",
+    "(dayofyear(date1) == 1)",
+    "(month(date1) == 1)",
+    # Date, Int
+    # AddMonths
+    "(add_months(date1, 1) == date1)",
```

Comment:
This filters everything out. How do we have confidence that it is doing the right thing, even in a non-UTC time zone?

Reply:
Updated.